### PR TITLE
Update hr_holidays_legal_leave errors. 

### PR DESCRIPTION
**Pull request by feketemihai**
Update traceback error with expected singleton on res.company in inverse method, fix comparing apples and oranges in compute method.

### CODE REVIEW

File: hr_holidays_legal_leave/models/hr_holidays_status.py [hunk 1]
```diff
@@ -26,10 +26,13 @@ def _get_default_company(self):
     @api.depends('company_id')
     def _compute_is_annual(self):
         for rec in self:
-            rec.is_annual = rec.company_id.legal_holidays_status_id == rec.id
+            legal_holiday = rec.company_id.legal_holidays_status_id
+            rec.is_annual = legal_holiday.id == rec.id
```
Comment (pedrobaeza) [inline hr_holidays_legal_leave/models/hr_holidays_status.py]: You can compare without the id:

```
rec.is_annual = legalholiday == rec
```

File: hr_holidays_legal_leave/models/hr_holidays_status.py [hunk 2]
```diff
@@ -26,10 +26,13 @@ def _get_default_company(self):
...
 
     @api.multi
     def _inverse_is_annual(self):
-        self.ensure_one()
-        if self.is_annual:
-            self.company_id.legal_holidays_status_id = self.id
+        for rec in self:
+            if not rec.company_id:
+                rec.company_id = self.env.user.company_id.id
```
Comment (pedrobaeza) [inline hr_holidays_legal_leave/models/hr_holidays_status.py]: You shouldn't auto-assign here.

File: hr_holidays_legal_leave/models/hr_holidays_status.py [hunk 3]
```diff
@@ -26,10 +26,13 @@ def _get_default_company(self):
...
+            if rec.is_annual:
```
Comment (pedrobaeza) [inline hr_holidays_legal_leave/models/hr_holidays_status.py]: add here the condition `and rec.company_id`.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]

File: hr_holidays_legal_leave/models/hr_holidays_status.py [hunk 4]
```diff
@@ -26,10 +26,10 @@ def _get_default_company(self):
...
+            rec.is_annual = rec.company_id.legal_holidays_status_id == rec
 
     @api.multi
     def _inverse_is_annual(self):
-        self.ensure_one()
-        if self.is_annual:
-            self.company_id.legal_holidays_status_id = self.id
+        for rec in self:
+            if rec.is_annual and rec.company_id:
+                rec.company_id.legal_holidays_status_id = rec.id
```
Comment (astirpe) [inline hr_holidays_legal_leave/models/hr_holidays_status.py]: `rec.company_id.legal_holidays_status_id = rec`

**Review by astirpe — CHANGES_REQUESTED** [review]

**Comment by feketemihai** [comment]
@OCA/human-resources-maintainers Can you have a look at this, please...the error is still present ...in v9 and v10...

**Review by astirpe — APPROVED** [review]
See my comment, is just a minor code review.
Anyway I approve this PR to avoid blocking this bugfix (needed).

**Comment by gurneyalex** [comment]
@feketemihai there seem to be a conflict in this PR. Could you rebase it please?

**Comment by feketemihai** [comment]
@gurneyalex Updated...thanks for the review...

**Review by dreispt — APPROVED** [review]

**Pull request merged**